scribus/text/storytext.cpp

```diff
@@ -198,21 +199,22 @@ void StoryText::moveCursorLeft(bool isGrapheme)
 {
 	if (isGrapheme)
 	{
-		int pos;
-		QTextBoundaryFinder boundaryFinder(QTextBoundaryFinder::Grapheme, plainText());
+		BreakIterator* it = getGraphemeIterator();
+		if (!it)
+			return;
+
+		int pos = cursorPosition();
+		it->setText(plainText().utf16());
 		if (paragraphStyle().direction() == ParagraphStyle::RTL)
 		{
-			boundaryFinder.setPosition(cursorPosition() + 1);
-			for (pos = 1; !boundaryFinder.isAtBoundary(); pos++)
-				boundaryFinder.setPosition(boundaryFinder.position() + 1);
+			pos = it->following(pos);
 		}
 		else
 		{
-			boundaryFinder.setPosition(cursorPosition() - 1);
-			for (pos = -1; !boundaryFinder.isAtBoundary(); pos--)
-				boundaryFinder.setPosition(boundaryFinder.position() - 1);
+			pos = cursorPosition();
```

Comment:
This is redundant, you already set `pos = cursorPosition()` before the if block.